torch/csrc/jit/python/init.cpp

```diff
@@ -885,13 +886,125 @@ void initJITBindings(PyObject* module) {
+                if (found_op) {
+                  auto check_and_append_overloaded_arg =
+                      [](std::vector<py::handle>& overloaded_args,
+                         const Argument& expected_arg,
+                         PyObject* arg) {
+                        if (THPVariable_CheckExact(arg)) {
+                          return;
+                        }
+                        if (check_has_torch_function(arg)) {
+                          append_overloaded_arg(overloaded_args, arg);
+                        } else if (expected_arg.type()->cast<ListType>()) {
+                          bool is_tuple = six::isTuple(arg);
+                          bool is_list = PyList_Check(arg);
+                          if (is_tuple || is_list) {
+                            size_t size = is_tuple ? PyTuple_GET_SIZE(arg)
+                                                   : PyList_GET_SIZE(arg);
+                            for (size_t idx = 0; idx < size; idx++) {
+                              PyObject* element = is_tuple
+                                  ? PyTuple_GET_ITEM(arg, idx)
+                                  : PyList_GET_ITEM(arg, idx);
+                              if (check_has_torch_function(element)) {
+                                append_overloaded_arg(overloaded_args, element);
+                              }
+                            }
+                          }
+                        }
+                      };
+                  const auto& expected_args = found_op->schema().arguments();
+                  for (size_t i = 0; i < expected_args.size(); ++i) {
+                    if (i < args.size()) {
+                      check_and_append_overloaded_arg(
+                          overloaded_args, expected_args[i], args[i].ptr());
+                    } else if (kwargs.contains(
+                                   expected_args[i].name().c_str())) {
+                      const auto& kwarg =
+                          kwargs[expected_args[i].name().c_str()];
+                      check_and_append_overloaded_arg(
+                          overloaded_args, expected_args[i], kwarg.ptr());
+                    }
+                    // Default value are only scalar or device types, which
+                    // don't support override.
+                  }
+                  if (overloaded_args.size() > 0) {
+                    std::vector<py::object> overloaded_types;
+                    overloaded_types.reserve(overloaded_args.size());
+                    for (auto& oarg : overloaded_args) {
+                      overloaded_types.push_back(
+                          py::reinterpret_borrow<py::object>(
+                              (PyObject*)Py_TYPE(oarg.ptr())));
+                    }
+                    py::tuple py_types = py::cast(overloaded_types);
+                    py::object ret;
+                    size_t delim_pos = op_name.find("::");
```

Comment:
You can use `Symbol` for parsing it (better yet, just pass `Symbol` which is just 4 bytes as a capture to this function instead of `op_name`)

There's `Symbol::toUnqualString()` for the second part and `Symbol::ns()::toUnqualString()` for the first

Reply:
Good point, forgot to use "Symbol"...